src/os2datascanner/projects/admin/adminapp/views/miniscanner_views.py

```diff
@@ -84,10 +79,35 @@ def execute_mini_scan(request):  # noqa:CCR001
                         if not message.matched:
                             continue
 
-                        replies.append(message)
+                        yield message
             else:
                 logger.warning(
                         "Miniscanner -"
                         " Rejected file that exceeded the size limit.")
 
+def execute_mini_scan(request):  # noqa:CCR001
+    context = {
+        "file_obj": (file_obj := request.FILES.get("file")),
+        "text": (text := request.POST.get("text")),
+        "raw_rule": (raw_rule := request.POST.get("rule")),
+        "halfbaked_rule": (halfbaked_rule := json.loads(raw_rule or "null")),
+
+        "replies": (replies := []),
+    }
+
+    rule = None
+    if halfbaked_rule:
+        rule = Rule.from_json_object(halfbaked_rule)
+
+    if file_obj:
+        for m in mini_scan(file_obj, rule):
+            replies.append(m)
+    if text:
+        for m in mini_scan(text, rule):
+            replies.append(m)
+
+<<<<<<< HEAD
```

Comment:
Looks like some git merge problems slipped through :laughing:

Reply:
They're gone now. Thanks for bringing it up !